diff --git a/ff/src/fields/models/fp/montgomery_backend.rs b/ff/src/fields/models/fp/montgomery_backend.rs
index a58c5ca0..c895b1f1 100644
--- a/ff/src/fields/models/fp/montgomery_backend.rs
+++ b/ff/src/fields/models/fp/montgomery_backend.rs
@@ -381,7 +381,7 @@ pub trait MontConfig<const N: usize>: 'static + Sync + Send + Sized {
                 r[(j + i) % N] =
                     fa::mac_with_carry(r[(j + i) % N], k, Self::MODULUS.0[j], &mut carry);
             }
-            r[i % N] = carry;
+            r[i] = carry;
         }
 
         BigInt::new(r)
